doc/developer/tooling.rst

```diff
@@ -2,39 +2,60 @@ Tooling and infrastructure
-``requirements.txt``, rather than ``setup.py``. ARMI itself has several requirements
-files, located at the root of the project, which can be used as jumping-off points for
-more complex scenarios.
+-----------------------------------
+The process of packaging Python projects and managing their dependencies is somewhat
+challenging and nuanced. The contents of our ``setup.py`` follow existing conventions as
+much as possible. In particular, we follow `this fantastic article
+<https://caremad.io/posts/2013/07/setup-vs-requirement/>`_ about dependecy management in
+Python projects.
+
+setup.py
+^^^^^^^^
+The packages listed in the ``install_requires`` argument to ``setup()`` are meant to
+express, as abstractly as possible, the packages that need to be installed **somehow**
+for the package to work. In addition, ``extras_require`` are used to specify other
+packages that are not strictly required, but if installed enable extra functionality,
+like unit testing or building documentation.
+
+requirements.txt
+^^^^^^^^^^^^^^^^
+The ``requirements***.txt`` files exist to describe a complete environment more
+specifically. If specific versions of packages are required, they should be defined here.
+Any extra arguments to ``pip`` will also be placed here. For instance, there is a ``-e``
+that tells ``pip`` to install ARMI itself and defer to ``setup.py`` for a version-agnostic
+list of dependencies. We also have multiple requirements files for different needs, like
+testing.
+
+Releasing a New Version of ARMI
+-------------------------------
+In ARMI, we use the common ``major.minor.bump`` version scheme where a version string
+might look like ``0.1.7``, ``1.0.0``, or ``1.2.123``. Each number has a specific meaning:
+
+* ``major`` - Revved for NRC-sanctioned release or at the end of a long development cycle.
+* ``minor`` - Revved when we decide the code or our API has reached a stable point;
+  this might happen once a year.
+* ``bump`` - Revved every time we modify the API, or at will; any time we want.
+
+**Any change to a major or minor version is considered a release.**
+
+Only a core member of the ARMI team may release a new version, or add a tag of any kind to
+the repository. The rule is *the only tags in the ARMI repo are for official versions*. If
+you want to release a version of ARMI, you will need admin privileges to multiple TerraPower
+repos on GitHub.
+
+Every release should follow this process:
+
+1. Ensure all unit tests pass and the documentation is building correctly.
+2. Add release notes to the documentation:
+   `here <https://github.com/terrapower/armi/tree/master/doc/release>`_.
+3. Tag the commit after it goes into the repo: ``git tag -a 1.0.0 -m "Release v1.0.0"``
+4. Also add the release notes on `the GitHub UI <https://github.com/terrapower/armi/releases>`_.
+5. Follow the instructions `here <https://github.com/terrapower/terrapower.github.io>`_ to
```

Comment:
these two `here` labels will be a sphinx warning :S so you should use anonymous links by making the trailing `_` actually be `__` (!)